### PR TITLE
GDEV-191 create dfs and fsx modules

### PR DESCRIPTION
**Pull request by syousry6**

### CODE REVIEW

File: aws/dfs/dfs.tf [hunk 1]
```diff
@@ -0,0 +1,443 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# VARIABLES / LOCALS / REMOTE STATE / AMI definition
+# ----------------------------------------------------------------------------------------------------------------------
+
+variable "map_migrated" {
```
Comment (jeremyodle) [inline aws/dfs/dfs.tf]: - [x] Remove MAP variables and use `var.tags` injection variable instead

File: aws/dfs/dfs.tf [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+  description = "common FSx ports"
+  vpc_id      = var.vpc_id
+  egress {
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "Egress"
+    from_port   = 0
+    protocol    = "-1"
+    to_port     = 0
+  }
+  tags = merge(local.tags,map(
```
Comment (jeremyodle) [inline aws/dfs/dfs.tf]: - [x] Remove MAP variables and use `var.tags` instead

File: aws/dfs/dfs.tf [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+    }
+  }
+
+  source_dest_check                    = length(var.network_interface) > 0 ? null : var.source_dest_check
+  disable_api_termination              = var.disable_api_termination
+  instance_initiated_shutdown_behavior = var.instance_initiated_shutdown_behavior
+  # placement_group                      = var.placement_group
+  tenancy = var.tenancy
+
+  tags = merge(
```
Comment (jeremyodle) [inline aws/dfs/dfs.tf:383]: - [x] Remove MAP variables and use `var.tags` instead

File: aws/dfs/module.tf [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+variable "name" {
+  type        = string
+  default     = "dfs"
+  description = "The name of the module"
+}
+
+variable terraform_module {
+  type        = string
+  default     = ""
```
Comment (jeremyodle) [inline aws/dfs/module.tf]: - [x] Set terraform_module to `gravicore/terraform-gravicore-modules/aws/dfs`

File: aws/dfs/module.tf [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    aws_region        = var.aws_region
+  }
+  automation_tags = {
+    terraform_module = var.terraform_module
+    stage_prefix     = local.stage_prefix
+    module_prefix    = local.module_prefix
+  }
+  security_tags = {}
+
+  map_tags = {
```
Comment (jeremyodle) [inline aws/dfs/module.tf]: - [x] Remove MAP tags and use `var.tags` instead

File: aws/fsx/module.tf [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+variable "name" {
+  type        = string
+  default     = "fsx"
+  description = "The name of the module"
+}
+
+variable terraform_module {
+  type        = string
+  default     = ""
```
Comment (jeremyodle) [inline aws/fsx/module.tf]: - [x] Set FSX terraform_module to `gravicore/terraform-gravicore-modules/aws/fsx`

**Review by jeremyodle — CHANGES_REQUESTED** [review]
- [x] Add modules to Labeler
https://github.com/gravicore/terraform-gravicore-modules/blob/master/.github/labeler.yml
- [x] Update README with new modules
- [x] Cleanup extra empty lines of code
- [x] Remove MAP variables/tags and use `var.tags` instead

File: aws/dfs/dfs.tf (hunk 2 above)
Comment (TysonBeck) [reply aws/dfs/dfs.tf]: This gets into some odd logic. In the module there is two instances which each need a different "map-migrated" Value. So this is passed in as a list. Tags cannot be lists, so it errors out. This is an override to remove it from the the map. I agree this does not make the module friendly. I think we should look into how we can apply this extra tag outside of the module in a supplemental file to help clean this up and keep it generic+

**Review by jeremyodle — DISMISSED** [review]

**Review by jeremyodle — APPROVED** [review]

**Pull request merged**